### DISCUSSION TITLE
Automatically select a remote kernel when opening a notebook

### DISCUSSION THREAD

**Discussion by jrich100** [microsoft/vscode-jupyter#13887] [Questions and Answers]
I have a kernel that is registered with a remote jupyter server and I want to use that kernel in every notebook.

The issue is that it takes several manual steps to select this kernel in the kernel selector, as well as having to first connect to the remote jupyter server. Ideally, this kernel would be automatically selected in every notebook on-open.

In previous versions (<=1.76), I was able to achieve this in a custom VS Code extension that would call `jupyter.selectjupyteruri` (now deprecated) using a known URI - I believe it would default to using the first kernel alphabetically, but worked for my use case.

Is there any way to make this process smoother now? Auto selecting a kernel, setting a default, or even just connecting to the remote jupyter server in an automated way?

@DonJayamanne it look like there are a lot of in-flight changes to the kernel selector, etc. but I wanted to see if there were any existing or planned ways to do this sort of thing. Thanks!

**Comment by DonJayamanne** [comment]
Please can you upvote this issue https://github.com/microsoft/vscode/issues/130946
